app/controllers/courses_controller.rb

```diff
@@ -24,6 +24,15 @@ def destroy
     head :no_content
   end
 
+  # get visible returns courses that are taught by an admin or enrolled in by a student
+  def get_visible
```

Comment:
I think we should just move this into users controller. Also return this as part of the `show` method.